File: Sliding window/121.cpp
#include <bits/stdc++.h>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, r = 1;
        int max_profit = 0;

        while (r < prices.size()) {
            if (prices[l] < prices[r]){
                max_profit = max(max_profit, prices[r] - prices[l]);
            } else {
                l = r;
            }
            r++;
        }

        return max_profit;
    }
}; 

File: Sliding window/121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let l = 0, r = 1;
    let max_profit = 0;

    while (r < prices.length){
        if (prices[l] < prices[r]){
            max_profit = Math.max(max_profit, prices[r] - prices[l]);
        } else {
            l = r;
        }
        r++;
    }
    return max_profit;
};